src/overlays/actors/ovl_En_In/z_en_in.c

```diff
@@ -33,75 +36,938 @@ const ActorInit En_In_InitVars = {
+            return 3;
+        }
+    }
+    return 0;
+}
+
+void func_80A79A2C(EnIn* this) {
+    s16 phi_v1;
+
+    if (this->unk_1EE != 3) {
+        if (this->unk_1EA == 0) {
+            phi_v1 = 0;
+        } else {
+            phi_v1 = --this->unk_1EA;
+        }
+        if (phi_v1 == 0) {
+            this->unk_1EE++;
+            if (this->unk_1EE >= 3) {
+                this->unk_1EA = Rand_S16Offset(30, 30);
+                this->unk_1EE = 0;
+            }
+        }
+    }
+}
+
+void func_80A79AB4(EnIn* this, GlobalContext* globalCtx) {
+    s32 i;
+    u32 f = 0;
+
+    if (this->skelAnime.animation != &D_06014CA8) {
+        f = globalCtx->gameplayFrames;
+    }
+    for (i = 0; i < ARRAY_COUNT(this->unk_330); i++) {
+        this->unk_330[i].y = (2068 + 50 * i) * f;
+        this->unk_330[i].z = (2368 + 50 * i) * f;
+    }
+}
+
+void func_80A79BAC(EnIn* this, GlobalContext* globalCtx, s32 index, u32 arg3) {
+    s16 entrances[] = { 0x0558, 0x04CA, 0x0157 };
+
+    globalCtx->nextEntranceIndex = entrances[index];
+    if (index == 2) {
+        gSaveContext.nextCutsceneIndex = 0xFFF0;
+    }
+    globalCtx->fadeTransition = arg3;
+    globalCtx->sceneLoadFlag = 0x14;
+    func_8002DF54(globalCtx, &this->actor, 8);
+    Interface_ChangeAlpha(1);
+    if (index == 0) {
+        AREG(6) = 0;
+    }
+    gSaveContext.timer1State = 0;
+}
+
+void func_80A79C78(EnIn* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    Vec3f sp48;
+    Vec3f sp3C;
+    Vec3s zeroVec = { 0, 0, 0 };
+
+    this->camId = Gameplay_CreateSubCamera(globalCtx);
+    Gameplay_ChangeCameraStatus(globalCtx, 0, 1);
+    Gameplay_ChangeCameraStatus(globalCtx, this->camId, 7);
+    sp48.x = this->actor.posRot.pos.x;
+    sp48.y = this->actor.posRot.pos.y + 60.0f;
+    sp48.z = this->actor.posRot.pos.z;
+    sp3C.x = sp48.x;
+    sp3C.y = sp48.y - 22.0f;
+    sp3C.z = sp48.z + 40.0f;
+    Gameplay_CameraSetAtEye(globalCtx, this->camId, &sp48, &sp3C);
+    this->actor.shape.rot.y = Math_Vec3f_Yaw(&this->actor.posRot.pos, &sp3C);
+    this->unk_308.unk_08 = zeroVec;
+    this->unk_308.unk_0E = zeroVec;
+    func_8010B680(globalCtx, 0x2025, NULL);
+    this->unk_308.unk_00 = 1;
+    player->actor.posRot.pos = this->actor.posRot.pos;
+    player->actor.posRot.pos.x += 100.0f * Math_SinS(this->actor.shape.rot.y);
+    player->actor.posRot.pos.z += 100.0f * Math_CosS(this->actor.shape.rot.y);
+    if (player->rideActor != NULL) {
+        player->rideActor->posRot.pos = player->actor.posRot.pos;
+        player->rideActor->freezeTimer = 10;
+    }
+    player->actor.freezeTimer = 10;
+    this->actor.flags &= ~1;
+    ShrinkWindow_SetVal(0x20);
+    Interface_ChangeAlpha(2);
+}
+
+s32 D_80A7B998 = 0;
+
+void EnIn_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnIn* this = THIS;
+    RespawnData* respawn = &gSaveContext.respawn[RESPAWN_MODE_DOWN];
+    Vec3f respawnPos;
+
+    this->objectIdx = Object_GetIndex(&globalCtx->objectCtx, 0xC0);
```

Comment:
Object enum can be used here, and I suggest renaming `objectIdx` to something a bit clearer, like `ingoObjBankIndex`
```suggestion
    this->objectIdx = Object_GetIndex(&globalCtx->objectCtx, OBJECT_IN);
```